### PR TITLE
[discovery-client] Spring cloud client basic authentication support

### PR DESCRIPTION
**Pull request by radovanradic**
Adding support for basic authentication configuration for spring cloud server.
Based on https://github.com/micronaut-projects/micronaut-discovery-client/issues/106 this is how I thought it could fixed.

### CODE REVIEW

**Comment by radovanradic** [comment]
@graemerocher I added another commit to fix the failing checks, did not realize initial commit was causing japiCmp task in the checks to fail. Don't know how to trigger checks now, thought commit would trigger them automatically.

**Comment by radovanradic** [comment]
Thanks @graemerocher 
All checks now passed except Static Analysis and it reports issue with sonarqube 
```
* What went wrong:
Execution failed for task ':sonarqube'.
> Project not found. Please check the 'sonar.projectKey' and 'sonar.organization' properties, the 'SONAR_TOKEN' environment variable, or contact the project administrator
```
which is probably unrelated to my changes.

File: discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudClientConfiguration.java [hunk 1]
```diff
@@ -113,6 +115,20 @@ public Optional<String> getName() {
...
+     */
+    public Optional<String> getUsername() {
+        return username != null ? Optional.of(username) : Optional.empty();
+    }
+
+    /**
+     * @return The spring cloud config password.
+     */
+    public Optional<String> getPassword() {
+        return password != null ? Optional.of(password) : Optional.empty();
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudClientConfiguration.java:129]: please change this to `Optional.ofNullable(password)` and remove the ternary operator

**Comment by alvarosanchez** [comment]
@radovanradic could please merge master into your branch to move this forward?

**Comment by graemerocher** [comment]
Rebased here https://github.com/micronaut-projects/micronaut-discovery-client/pull/366

**Pull request closed without merging**